src/System.Security.Cryptography.Pkcs/ref/System.Security.Cryptography.Pkcs.netcoreapp.cs

```diff
@@ -5,8 +5,72 @@
 // Changes to this file must follow the http://aka.ms/api-review process.
 // ------------------------------------------------------------------------------
 
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+
 namespace System.Security.Cryptography.Pkcs
 {
+    public enum Rfc3161RequestResponseStatus
+    {
+        Unknown = 0,
+        Accepted = 1,
+        DoesNotParse = 2,
+        RequestFailed = 3,
+        HashMismatch = 4,
+        VersionTooNew = 5,
+        NonceMismatch = 6,
+        RequestedCertificatesMissing = 7,
+        UnexpectedCertificates = 8,
+    }
+    public sealed partial class Rfc3161TimestampRequest
+    {
+        private Rfc3161TimestampRequest() { }
+        public int Version => throw null;
+        public ReadOnlyMemory<byte> GetMessageHash() => throw null;
+        public Oid HashAlgorithmId => throw null;
+        public Oid RequestedPolicyId => throw null;
+        public bool RequestSignerCertificate => throw null;
+        public ReadOnlyMemory<byte>? GetNonce() => throw null;
+        public bool HasExtensions => throw null;
+        public X509ExtensionCollection GetExtensions() => throw null;
+        public Task<Rfc3161TimestampToken> SubmitRequestAsync(Uri uri, TimeSpan timeout) => throw null;
```

Comment:
It looks like this will throw whenever a non-zero, non-one timeStampStatus is returned. Is that a generally exceptional scenario, or should we instead have a `Rfc3161TimestampResponse` type that contains the timeStampStatus and the TimeStampToken (if present) to allow custom handling of the other cases?

Reply:
The token is only present on 0 (issued) or 1 (issued with changes).  I didn't encounter any timestamp servers that issued any of the other codes.  (And most programs wouldn't handle "I don't have an answer yet" very well, anyways).  We could add the other functionality later, if needed.

Reply:
Nevermind, I missed the "TryAcceptResponse" method that handles that.

Reply:
Though in that case you don't get the Uri overload and async pattern. Which I guess is fine.